iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractTapChanger.java

```diff
@@ -7,23 +7,24 @@
 package com.powsybl.iidm.network.impl;
 
 import com.powsybl.commons.util.trove.TBooleanArrayList;
-import com.powsybl.iidm.network.Terminal;
-import com.powsybl.iidm.network.ValidationException;
-import com.powsybl.iidm.network.ValidationUtil;
+import com.powsybl.iidm.network.*;
 import com.powsybl.iidm.network.impl.util.Ref;
 import gnu.trove.list.array.TDoubleArrayList;
 import gnu.trove.list.array.TIntArrayList;
 
 import java.util.List;
 import java.util.Objects;
+import java.util.OptionalInt;
 
 /**
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 abstract class AbstractTapChanger<H extends TapChangerParent, C extends AbstractTapChanger<H, C, S>, S extends TapChangerStepImpl<S>> implements MultiVariantObject {
 
-    protected final Ref<? extends VariantManagerHolder> network;
+    protected final NetworkImpl network;
```

Comment:
The purpose of Ref was to allow breaking revert link to network in case of merge. I think here you added a regression and this is redundant with networkRef which already allows you to get network in a safe manner.

Reply:
Same remark everywhere in the rest of the PR